### PR TITLE
add expiration check

### PR DESCRIPTION
**Pull request by chack05**
- zones: only delivered if not expired
- messages: only delivered if not expired and the referring zone is not expired as well

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -117,8 +150,8 @@ server.post('/api/addmessages', function(req, res) {
     var msgTable = nano.use(msgdb);
 
     var error = null;
-    for (let i = 0; i < req.body.Messages.length; i++) {
-        let message = req.body.Messages[i];
+    for (let mCount = 0; mCount < req.body.Messages.length; mCount++) {
```
Comment (JanVan01) [inline index.js]: Why did you change that? That twas a totally valid way of writing a for-loop. Or did I miss something?

File: index.js [hunk 2]
```diff
@@ -43,8 +43,12 @@ server.get('/api/zones', function(req, res) {
         include_docs: true
     }, function(err, body, header) {
         if (!err) {
-            for (var i = 0; i < body.rows.length; i++) {
-                zones.Zones.push(body.rows[i].doc);
+            for (var zCount = 0; zCount < body.rows.length; zCount++) {
+                let zoneDate = new Date(body.rows[zCount].doc["Expired-at"]);
+                let nowDate = new Date();
+                if(zoneDate > nowDate.getTime()) {
+                    zones.Zones.push(body.rows[zCount].doc);
```
Comment (JanVan01) [inline index.js]: I know this works, but we are still doing database work here. It is time to do that in the database. We can't add more and more code here that we will have to change at some point again.

File: index.js (hunk 1 above)
Comment (chack05) [inline index.js]: 'i' is a non-descriptive variable name.

File: index.js [hunk 3]
```diff
@@ -90,20 +95,48 @@ server.get('/api/messages', function(req, res) {
...
+                    for (let mCount = 0; mCount < mbody.rows.length; mCount++) {
+                        // filter the zone id
+                        if (mbody.rows[mCount].doc["Zone-id"] == parseInt(req.query.zone)) {
+                            //check if the message expired
+                            let msgDate = new Date(mbody.rows[mCount].doc["Expired-at"]);
+                            let nowDate = new Date();
+                            if(msgDate > nowDate.getTime()) {
+                            
+                                // check if the zone expired    
+                                for (let zCount = 0; zCount < zbody.rows.length; zCount++) {
```
Comment (JanVan01) [inline index.js]: Why are you using one for loop in an other this will take a lot of time...
You could solve this better by fist calling the zonesdb to check if the zone is valid and then if it is call the messagesdb and select the messages belonging to the zone.
And see my comment from above concerning database work.

File: index.js (hunk 1 above)
Comment (JanVan01) [inline index.js]: Well it's a for loop iterating over indices it is kind of descriptive ;-P

Comment (chack05) [inline index.js]: No it's crap, ever. In particular as soon as you have nested loops or bigger loop bodies.

My mistake of introducing it, but now it's gone.

File: index.js (hunk 3 above)
Comment (chack05) [inline index.js]: That's true, but since I was kind in a hurry, I didn't get to do that the nice way.

Comment (JanVan01) [inline index.js]: Just don't nest them and it takes linear time.

File: index.js (hunk 2 above)
Comment (chack05) [inline index.js]: We don't even have a strategy for how the views are getting in the database.

I asked Salman to look into that, but he [posted](http://www.barkingiguana.com/2009/01/20/adding-a-simple-view-to-couchdb/) some link. We need to take into account first how the views initially get from Node to the CouchDB.

File: index.js (hunk 3 above)
Comment (chack05) [inline index.js]: Fair enough, I probably forgot that the messages can only belong to one zone at that point. I might do this later.

Regarding the DB views, I think we should spend a meeting for this.

Comment (chack05) [inline index.js]: @JanVan01 Done. Please review and if it's OK merge.

Once that's done I probably could start looking into creating and integrating json schemas.
